### PR TITLE
deflake orphaning GC

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/14353

@sttts @liggitt @mfojtik @ironcladlou  I'm open to better suggestions.  I described the issue in the code.  Longer term, perhaps we have an LRU of orphanings?

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

**Comment by deads2k** [comment]
Can I get someone to go on record one way or the other?  This one would be worthy of a green button to deflake us.

cc @eparis

**Comment by liggitt** [comment]
LGTM

File: test/integration/gc_default_test.go [hunk 1]
```diff
@@ -71,6 +71,11 @@ func TestGCDefaults(t *testing.T) {
 	if _, err := kubeClient.Core().ConfigMaps(ns).Create(childConfigMap); err != nil {
 		t.Fatal(err)
 	}
+	// we need to make sure that the GC graph has observed the creation of the configmap *before* it observes the delete of
+	// the buildconfig or the orphaning step won't find anything to orphan, then the delete will complete, the configmap
+	// creation will be observed, there will be no parent, and the configmap will be deleted.
+	// There is no API to determine if the configmap was observed.
```
Comment (kargakis) [inline test/integration/gc_default_test.go]: This seems worth discussing upstream.

**Pull request merged**